onnxruntime/contrib_ops/cuda/quantize_ops.cc

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cuda_common.h"
+#include "core/providers/cuda/tensor/quantize_linear.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+using namespace onnxruntime::cuda;
+
+#define REGISTER_Q_KERNEL_TYPED(T, U)                              \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                   \
+      QuantizeLinear,                                              \
```

Comment:
Note that for the CPU version of QuantizeLinear, there was a difference in the rounding rules between the com.microsoft and ONNX versions. The com.microsoft had been implemented with std::round or round away from zero mode while the ONNX version explicitly says it is round half to even.

Ideally, the CPU version would also just use round half to even too and then the CPU EP could just use MlasQuantizeLinear to implement the inner loop instead. This would change model results though that used the op before (from WinML days).